src/main/java/it/smartcommunitylab/aac/webauthn/controller/ControllerUtils.java

```diff
@@ -0,0 +1,41 @@
+package it.smartcommunitylab.aac.webauthn.controller;
+
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import javax.servlet.http.HttpSession;
+
+import org.springframework.web.context.request.RequestContextHolder;
+import org.springframework.web.context.request.ServletRequestAttributes;
+
+public class ControllerUtils {
```

Comment:
this class should go. Put the methods back where you use them and then we'll review from there

Reply:
Moved methods for username validation to a new class, `src/main/java/it/smartcommunitylab/aac/webauthn/controller/UsernameValidator.java`.
Its purpose is to validate the constraints we can impose on the username and the display name (e.g., it must be between 3 and 30 characters, with no special symbols...)